tonic/src/codec/compression.rs

```diff
@@ -104,13 +137,19 @@ impl CompressionEncoding {
         let header_value = map.get(ACCEPT_ENCODING_HEADER)?;
         let header_value_str = header_value.to_str().ok()?;
 
-        split_by_comma(header_value_str).find_map(|value| match value {
-            #[cfg(feature = "gzip")]
-            "gzip" => Some(CompressionEncoding::Gzip),
-            #[cfg(feature = "zstd")]
-            "zstd" => Some(CompressionEncoding::Zstd),
-            _ => None,
+        // Get the highest supported encoding and its priority
+        split_by_comma(header_value_str).filter_map(|value| {
+            match value {
+                #[cfg(feature = "gzip")]
+                "gzip" => Some(CompressionEncoding::Gzip),
+                #[cfg(feature = "zstd")]
+                "zstd" => Some(CompressionEncoding::Zstd),
+                _ => None,
+            }.map(|encoding| (encoding, enabled_encodings.priority(encoding)))
         })
+            .take(COMPRESSION_ENCODINGS_LENGTH)
+            .max_by_key(|(_, priority)| priority.unwrap_or(usize::MIN))
```

Comment:
Isn't usize::MIN 0?

Reply:
for now, yeah, but what if it changes?

Reply:
I meant, is it correct here? You are basically saying that if an encoding doesn't appear in the priority list, it has the same priority as the first one in the list.

Reply:
good point... changed to exclude, and changed to only take at the start to help prevent dos